File: src/main/java/com/whatever/entity/Player.java
package com.whatever.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Player {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    
    @NotNull(message = "Nome não pode ser vazio")
    @Column(unique = true)
    private String nome;
    
    private int partidas;
    
    private int pontuacao;

}

File: src/main/resources/static/front/src/app/models/player.ts
export class Player {

    id!: number;
    nome!: string;
    partidas!: number;
    pontuacao!: number;

    constructor(id: number, nome: string, partidas: number, pontuacao: number){
        this.id = id;
        this.nome = nome;
        this.partidas = partidas;
        this.pontuacao = pontuacao;
    }

}
